src/drivers/fluid_winmidi.c

```diff
@@ -97,8 +153,12 @@ fluid_winmidi_callback(HMIDIIN hmi, UINT wMsg, DWORD_PTR dwInstance,
 
     case MIM_DATA:
         event.type = msg_type(msg_param);
-        event.channel = msg_chan(msg_param);
+        event.channel = msg_chan(msg_param) + dev_infos->channel_map;
 
+#ifdef PRINTF_MSG
+        printf("\ndev_idx:%d, channel in:%d,out: %d\n",
```

Comment:
I think it would be worth to convert those `printf`s to `FLUID_LOG(FLUID_DBG)` messages + a little more user-friendly message of what's going on. Otherwise, I don't see a benefit of having those `printf`s in the code.

Reply:
Ok.